### ISSUE TITLE
Visual Identifier Expansion inside Vim Shell

### ISSUE THREAD

**Issue by philFernandez** [romkatv/powerlevel10k#359]
TLDR; How can I have the prompt be aware that I opened a shell from vim, and change its appearance when I'm inside a shell session that was started inside vim?

I moved over to powerlevel10k from powerlevel9k and am enjoying the performance increase. I was using my old powerlevel9k config file for a while, and everything worked out of the box, just faster. I recently disabled my config and set up a new one with the automated tool, which was a great experience. I just have one little issue that I'm trying to work out.

I use vim, and oftentimes I drop into a shell from vim by running :sh inside vim's command mode. This is similar to pressing 'S' while inside ranger. It just opens a shell while vim is still running in the background. When I quit the shell it drops me back into vim. 

The theme that resulted from running the automatic tool gave me a feature where the right side prompt lets me know when I'm in a shell that was generated by ranger. I'm trying to add similar functionality for vim. With my old config I was able to have 
`if [ $VIMRUNTIME ]; then
      POWERLEVEL9K_CONTEXT_TEMPLATE=$'\Ue7c5 %m';
 fi
`
which added a little vim icon to the prompt when inside a shell that was opened from vim. I tried adding this option to the generated ~/.p10k file, but it throws errors when loading up. How can I get similar functionality to that which happens when opening a shell from ranger for vim?

**Comment by philFernandez** [comment]
I figured it out..

**Comment by romkatv** [comment]
I implemented `vim_shell` prompt segment and enabled it by default. If you update powerlevel10k and create a new config with `p10k configure`, you'll get an icon in your prompt whenever you start zsh from vim.

**Comment by philFernandez** [comment]
> I implemented `vim_shell` prompt segment and enabled it by default. If you update powerlevel10k and create a new config with `p10k configure`, you'll get an icon in your prompt whenever you start zsh from vim.

I implemented it almost exactly the same in the meantime yesterday locally on my machine. I guess I should have submitted a PR when I finally figured out. Anyway, I'll update now and use yours so I can stay even with the master branch here. Thank you for the awesome plugin!
